plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -677,6 +679,24 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         return Optional.of(new JdbcTypeHandle(Types.NUMERIC, Optional.of("decimal"), Optional.of(decimalType.getPrecision()), Optional.of(decimalType.getScale()), Optional.empty(), Optional.empty()));
     }
 
+    @Override
+    public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<SortItem> sortOrder)
+    {
+        Map<String, JdbcColumnHandle> columns = getColumns(session, handle).stream()
+                .collect(toImmutableMap(JdbcColumnHandle::getColumnName, identity()));
+
+        // PostgreSQL by default orders lowercase letters before uppercase, which is different from Trino
+        // TODO We could still push the sort down if we could inject a PostgreSQL-specific syntax for selecting a collation for given comparison.
+        boolean disablePushdown = sortOrder.stream()
+                .anyMatch(sortItem -> {
+                    verify(columns.containsKey(sortItem.getName()));
+                    Type sortItemType = columns.get(sortItem.getName()).getColumnType();
+                    return sortItemType instanceof CharType || sortItemType instanceof VarcharType;
+                });
+
+        return !disablePushdown && super.supportsTopN(session, handle, sortOrder);
```

Comment:
Don't call `super`. It is not going to provide any more information than we have.

Reply:
Removed.